capture/lib/ot_communication.py

```diff
@@ -3,86 +3,223 @@
 # SPDX-License-Identifier: Apache-2.0
 """Communication interface for different OpenTitan ciphers.
 
-Currently, communication with each cipher happens over simpleserial. This file
-provides communication interface classes for each of these ciphers.
+Communication with OpenTitan either happens over simpleserial or the uJson
+command interface.
 """
+import json
+import time
+from typing import Optional
+
+import serial
+
+
+class OTUART:
+    def __init__(self, protocol: str, port: Optional[str] = "None",
+                 baudrate: Optional[int] = 115200) -> None:
+        self.uart = None
+        if protocol == "ujson":
+            if not port:
+                raise RuntimeError("Error: No uJson port provided!")
+            else:
+                self.uart = serial.Serial(port)
+                self.uart.baudrate = baudrate
 
 
 class OTAES:
-    def __init__(self, target) -> None:
+    def __init__(self, target, protocol: str,
+                 port: Optional[OTUART] = None) -> None:
         self.target = target
+        self.simple_serial = True
+        self.port = port
+        if protocol == "ujson":
+            self.simple_serial = False
+            # Init the AES core.
+            self.port.write(json.dumps("AesSca").encode("ascii"))
+            self.port.write(json.dumps("Init").encode("ascii"))
 
-    def write_key(self, key):
+    def _ujson_aes_sca_cmd(self):
+        # TODO: without the delay, the device uJSON command handler program
+        # does not recognize the commands.
+        time.sleep(0.01)
+        self.port.write(json.dumps("AesSca").encode("ascii"))
+
+    def key_set(self, key: list[int], key_length: Optional[int] = 16):
         """ Write key to AES.
         Args:
             key: Bytearray containing the key.
         """
-        self.target.simpleserial_write("k", key)
+        if self.simple_serial:
+            self.target.simpleserial_write("k", bytearray(key))
+        else:
+            # AesSca command.
+            self._ujson_aes_sca_cmd()
+            # KeySet command.
+            self.port.write(json.dumps("KeySet").encode("ascii"))
+            # Key payload.
+            time.sleep(0.01)
+            key_data = {"key": key, "key_length": key_length}
+            self.port.write(json.dumps(key_data).encode("ascii"))
 
-    def fvsr_key_set(self, key):
+    def fvsr_key_set(self, key, key_length: Optional[int] = 16):
```

Comment:
It looks like this function is not used anywhere.

Reply:
Yes, you are right. However, as the AES SCA uJSON handler at the OT device implements the _FvsrKeySet_ command, the handler is provided here. If we don't need the _FvsrKeySet_ command at the OT device, we should remove it there and also here.